cpi/pinocchio/src/lib.rs

```diff
@@ -0,0 +1,60 @@
+//! Rust example demonstrating invoking another program
+#![deny(missing_docs)]
+
+use pinocchio::{
+    account_info::AccountInfo,
+    entrypoint,
+    entrypoint::ProgramResult,
+    instruction::{Account, AccountMeta, Instruction, Seed, Signer},
+    program::invoke_signed_unchecked,
+    program_error::ProgramError,
+    pubkey::{create_program_address, Pubkey},
+};
+
+pinocchio::entrypoint!(process_instruction);
+
+/// Amount of bytes of account data to allocate
+pub const SIZE: usize = 42;
+
+/// Instruction processor
+pub fn process_instruction(
+    program_id: &Pubkey,
+    accounts: &[AccountInfo],
+    instruction_data: &[u8],
+) -> ProgramResult {
+    // Account info to allocate abd for the program being invoked
+    let [allocated_info, _system_program_info] = accounts else {
+        return Err(ProgramError::NotEnoughAccountKeys);
+    };
+
+    let expected_allocated_key =
+        create_program_address(&[b"You pass butter", &[instruction_data[0]]], program_id)?;
+    if *allocated_info.key() != expected_allocated_key {
+        // allocated key does not match the derived address
+        return Err(ProgramError::InvalidArgument);
+    }
+
+    // Invoke the system program to allocate account data
+    let mut data = [0; 12];
+    data[0] = 8; // ix discriminator
+    data[4..12].copy_from_slice(&SIZE.to_le_bytes());
```

Comment:
I wonder if the compiler knows to make this into a static array, or if you'll gain a few more CUs by just having a const of `[8, 0, 0, 0, 42, 0, 0, 0, 0, 0, 0, 0]`

Reply:
This did not make a difference, probably already being optimized that way.